src/abstractdataframe/abstractdataframe.jl

```diff
@@ -983,35 +992,78 @@ julia> vcat(df1, df2)
 │ 6   │ 6     │ 6     │
 ```
 """
-Base.vcat(df::AbstractDataFrame) = df
-Base.vcat(dfs::AbstractDataFrame...) = _vcat(collect(dfs))
-function _vcat(dfs::AbstractVector{<:AbstractDataFrame})
+Base.vcat(df::AbstractDataFrame; 
+          columns::Union{Symbol, AbstractVector{Symbol}}=:equal) = df
+
+Base.vcat(dfs::AbstractDataFrame...;
+          columns::Union{Symbol, AbstractVector{Symbol}}=:equal) = 
+    _vcat(collect(dfs); columns=columns)
+
+function _vcat(dfs::AbstractVector{<:AbstractDataFrame}; 
+               columns::Union{Symbol, AbstractVector{Symbol}}=:equal)
     isempty(dfs) && return DataFrame()
+    # Array of all headers
     allheaders = map(names, dfs)
+    # Array of unique headers across all data frames
     uniqueheaders = unique(allheaders)
+    # All symbols present across all headers
     unionunique = union(uniqueheaders...)
+    # List of symbols present in all dataframes
     intersectunique = intersect(uniqueheaders...)
-    coldiff = setdiff(unionunique, intersectunique)
-
-    if !isempty(coldiff)
-        # if any DataFrames are a full superset of names, skip them
-        filter!(u -> Set(u) != Set(unionunique), uniqueheaders)
-        estrings = Vector{String}(undef, length(uniqueheaders))
-        for (i, u) in enumerate(uniqueheaders)
-            matching = findall(h -> u == h, allheaders)
-            headerdiff = setdiff(coldiff, u)
-            cols = join(headerdiff, ", ", " and ")
-            args = join(matching, ", ", " and ")
-            estrings[i] = "column(s) $cols are missing from argument(s) $args"
+
+    # Throw an informative error if names arent the same across all dataframes
+    if columns === :equal
+            header = unionunique
+            coldiff = setdiff(unionunique, intersectunique)
```

Comment:
it is probably enough to compare length.

Reply:
Yeah I agree this error message is overkill. Particularly because I can't imagine someone `vcat`ing a long enough array of vectors to get confused. But it was here when I started the PR.